kivymd/theming.py

```diff
@@ -1052,7 +1053,18 @@ def __init__(self, **kwargs):
         Clock.schedule_once(lambda x: self.on_theme_style(0, self.theme_style))
         self._determine_device_orientation(None, Window.size)
         Window.bind(size=self._determine_device_orientation)
+        self.bind(font_styles=self.sync_theme_styles)
         self.colors = colors
+        Clock.schedule_once(self.sync_theme_styles)
+
+    def sync_theme_styles(self, *args):
+        # Syncs the values from self.font_styles to theme_font_styles
+        # this will ensure continuity when someone registers a new font_style.
+        for num, style in enumerate(theme_font_styles):
+            if style not in self.font_styles:
+                theme_font_styles.pop(num)
+        for style in self.font_styles.keys():
+            theme_font_styles.append(style)
```

Comment:
See comment below.

To me - this looks like a complex solution to fix a problem that does not really exist.

Reply:
Both namespaces should keep the same keys, values, otherwise, why should it exists as separated elements?